core/Scheduler/Scheduler.php

```diff
@@ -147,6 +152,33 @@ public function run()
                     $readFromOption = true;
                     $message = $this->executeTask($task);
 
+                    // Task has thrown an exception and should be scheduled for a retry
+                    if ($this->scheduleRetry) {
+
+                        if($this->timetable->getRetryCount($task->getName()) == 3) {
+
+                            // Task has already been retried three times, give up
+                            $this->timetable->clearRetryCount($task->getName());
+
+                            $this->logger->info("Scheduler: '{task}' has already been retried three times, giving up",
```

Comment:
Shouldn't we at least log a warning in that case, so we still notice tasks that keep failing and only random errors are hidden?